CONTRIBUTING.md

```diff
@@ -86,30 +190,80 @@ devs:
 - `g_strsplit()`
 - `g_strcmp0`
 
-## Commit Messages
+# The use of GNU extensions
+
+We avoid [GNU C extensions] because we want to fit into the eco-system
+(wayland and wlroots) we live in.
+
+We do use `__typeof__` which strictly speaking is a GNU C extension (`typeof`)
+but through the use of `__` is supported by gcc and clang without defining
+`_GNU_SOURCE`. The justification for this is that Wayland uses it, for example
+in the [`wl_container_of()`] macro which is needed in `wl_list*` and it
+does provide pretty big benefits in terms of type safety.
+
+We compile with `-std=c11` because that's what 'wlroots' and we do not want
+to increase the entry-level for OSs without good reason (and currently we
+can't think of one).
+
+# Commit Messages
+
+The log messages that explain changes are just as important as the changes
+themselves. Try to describe the 'why' to help future developers.
 
 Write [commit messages] like so, keeping the top line to this sort of syntax:
 
 ```
 cursor: add special feature
 ```
 
-And please wrap the main commit message at max 74 characters, otherwise `git log` and similar look so weird.
+This first line should:
+
+- Be a short description
+- In most cases be prefixed with "area: " where area refers to a filename
+  or identifier for the general area of the code being modified.
+- Not capitalize the first word following the "area: " prefix, unless
+  it's a name, acronym or similar.
+- Skip the full stop
 
-## Naming Convention
+And please wrap the main commit message at max 74 characters, otherwise
```

Comment:
Just for me to understand, "main commit message" == "first line"?

Reply:
I mean the whole thing. I'll remove the word "main" to make it clearer.
Some projects seem to have a soft limit on the first line of r50-52... but I don't really understand the rationale for this and often find it can be a challenge to write a sensible first line within that tight constraint.